SportsHub/UnitTests/Services/TestUserService.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SportsHub.AppService.Services;
+using SportsHub.DAL.Data;
+using SportsHub.DAL.UOW;
+using SportsHub.Domain.UOW;
+using Xunit;
+
+namespace UnitTests.Services;
+
+public class TestUserService : IDisposable
+{
+    private readonly ApplicationDbContext _context;
+    private readonly IUnitOfWork _unitOfWork;
+
+    public TestUserService()
+    {
+        var options = new DbContextOptionsBuilder<ApplicationDbContext>()
+            .UseInMemoryDatabase(databaseName: Guid.NewGuid().ToString())
+            .Options;
+        _context = new ApplicationDbContext(options);
+        _context.Database.EnsureCreated();
+        _context.Users.AddRange(MockData.UsersMockData.GetUsers());
+        _unitOfWork = new UnitOfWork(_context);
+        _unitOfWork.SaveChangesAsync();
+    }
+
+    [Theory]
+    [InlineData("firstuser@mail.com")]
+    [InlineData("seconduser@mail.com")]
+    public async Task GetByEmailAsync_WithCorrectEmail_ReturnsCorrectUser(string email)
+    {
+        var sut = new UserService(_unitOfWork);
```

Comment:
create sut in constructor, there no specific configs for every test

Reply:
I left it there because the sut is my arrange and i wanted to be more visible . Is it good practice when the arrange is the same for all the tests in the class to move it in the constructor and leave only act and assert in the test